src/averageAges.js

```diff
@@ -14,12 +14,21 @@
  *
  * @return {number}
  */
+const MALE = 'm';
+const FEMALE = 'f';
+const AGE_DIVIDER = 100;
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const manOnly = people.filter((person) => {
+    return !century
+      ? person.sex === MALE
+      : century === Math.ceil(person.died / AGE_DIVIDER) && person.sex === MALE;
+  });
+
+  const sumManAge = manOnly.reduce((sumAge, person) =>
+    sumAge + (person.died - person.born), 0);
+
+  return +(sumManAge / manOnly.length).toFixed(2);
```

Comment:
Remember about DRY. You have the same logic in the second function. Move this part to separate function